### ISSUE TITLE
Multimodule project dependencies in a different layer

### ISSUE THREAD

**Issue by loosebazooka** [GoogleContainerTools/jib#1724]
For the purpose of this bug, we will call maven module and gradle sub-project dependencies "modules".

### Current Layers
* classes
* resources
* dependencies (with modules that are not versioned `SNAPSHOT`)
* snapshot-dependencies (with module versioned as `SNAPSHOT`)

### Proposed Layers (1): All module deps should be considered snapshots
* classes
* resources
* dependencies (no module deps)
* snapshot-dependencies (includes all module deps)

But this can just be achieved by versioning any module as SNAPSHOT, so maybe there's no value in (1)

### Proposed Layers (2): Module deps are their own layer
* classes
* resources
* dependencies (no module deps)
* snapshot-dependencies (no module deps)
* module dependencies (all module deps)

@GoogleContainerTools/java-tools-build What do you think? (2) sounds good to me

**Comment by chanseokoh** [comment]
Both make sense to me. Slight preference for (2). I vaguely remember someone was saying along the line snapshot dependencies and module dependencies may not be at the same level, but I may be wrong.

This is to fix #1436.

**Comment by loosebazooka** [comment]
Okay so I guess here are my proposals for updating the JavaImageBuilder api, given that we're probably going with style (2) from [above](https://github.com/GoogleContainerTools/jib/issues/1724#issue-445582121).

```java
JavaContainerBuilder.addDependencies(List<Path> dependencies)
```

will become

1.   
    ```java
    JavaContainerBuilder.addDependencies(List<Path> dependencies) // fixed version dependencies
    JavaContainerBuilder.addSnapshotDependencies(List<Path> dependencies) //snapshot or changing dependencies
    JavaContainerBuilder.addProjectDependencies(List<Path> dependencies) //subproject/module dependencies
   ```

or 

2. This is a little more complicated, and the more I think about it, the less I like it.
    ```
    JavaContainerBuilder.setDependencies(DependencyProvider dependencies)
   ```

**Comment by chanseokoh** [comment]
I like (1). Hopefully this covers most of the use cases.

**Comment by loosebazooka** [comment]
Yeah this is opinionated, so I think it'll be fine.

**Comment by TadCordle** [comment]
Yeah, I like 1 as well.

**Comment by chanseokoh** [comment]
Fixed by #1780 and #1785, enabled by #1773.
